fb/xrgb.go

```diff
@@ -0,0 +1,109 @@
+// Copyright ©2016 The ev3go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package fb
+
+import (
+	"errors"
+	"image"
+	"image/color"
+	"image/draw"
+)
+
+// NewXRGB returns a new XRGB image with the given bounds.
+func NewXRGB(r image.Rectangle) *XRGB {
+	w, h := r.Dx(), r.Dy()
+	stride := 2 * w
+	pix := make([]uint8, stride*h)
+	return &XRGB{Pix: pix, Stride: stride, Rect: r}
+}
+
+// NewXRGBWith returns a new XRGB image with the given bounds,
+// backed by the []byte, pix. If stride is zero, a working stride
+// is computed. If the length of pix is less than stride*h, an
+// error is returned.
+func NewXRGBWith(pix []byte, r image.Rectangle, stride int) (draw.Image, error) {
+	w, h := r.Dx(), r.Dy()
+	if stride == 0 {
+		stride = 4 * w
+	}
+	if len(pix) < stride*h {
+		return nil, errors.New("ev3dev: bad pixel buffer length")
+	}
+	return &XRGB{Pix: pix, Stride: stride, Rect: r}, nil
+}
+
+// XRGB is an in-memory image whose At method returns Pixel565 values.
```

Comment:
It looks like the At method returns color.Color, not Pixel565

Reply:
This is fixed, but `At` has to return `color.Color` to satisfy the `image.Image` interface. The underlying concrete value is a `PixelXRGB`.